### DISCUSSION TITLE
About the herbarium system

### DISCUSSION THREAD

**Discussion by dadiorchen** [Greenstand/treetracker-decisions#30] [Ideas]
We have discussed the herbarium in these two PRs:
https://github.com/Greenstand/treetracker-decisions/pull/21
https://github.com/Greenstand/treetracker-decisions/pull/14

I think they both output the same thing: build two libraries: a global standard species library and a native/local organization species library and build up the relationship between them. The difference is that PR #21 uses an extra determination rule to make the final decision.

So the first thing we need to discuss and decide is that do we need this comprehensive determining system. I tend to say yes, at least, we should RECORD down all these determining actions on the DB, so we can still change the rules and evolve a good herbarium system in the further. 

If we choose the direction of #21 , to the rule described by @camwebb that uses the latest record as the final determination, I have another idea, is to use the role system to add weight to users, we give users different roles, say, organization-operator, species-editor, species-expert, and attach different weight to the role, (which we can easily implement by Keycloak without coding) so the final determination could be decided by rules that involve the factors like: weight, timing and so on.

So if we decided to go for #21 I can give a complete design based on #21 and give some UX Figma so we can push forward this project.

**Comment by sebastiangaertner** [comment]
Agree with 21 and we should be building a new team around this.

**Reply by lorepostman** [reply]
This is interesting (and exciting)! I'll watch for grant opportunities to fund this project specifically. One suggestion/request: as you build the team requirements, please try to outline all of the talent that would be needed to get this work to mvp within an 18 month timeframe. Time bound grant entries tend to do better. Also be generous in the amount of talent this will take. It's easier to sell a big budget at the start than to try to get extra money midway through.

**Comment by camwebb** [comment]
Glad to see this going ahead. The design in [0012-species-data.md](https://github.com/Greenstand/treetracker-decisions/blob/main/0012-species-data.md) still stands. 

The expertise system suggested by @dadiorchen is a good idea. Remember there will be two levels of determination: i) the application of a local name by members of the planting org, and ii) the mapping of a local name to a global (within GS) name.  The former may involve several users but the choices will be quite limited, because each organization generally knows its own species' local names well, even if they do not know scientific names. The latter (ii) is the tricky part, and requires botanical experience. This has always been the bottle-neck (me, so far), and will continue to be. @lorepostman, the level of engagement required by an experienced botanist really requires funding. I've done a fair amount as a volunteer, but simply can't devote much time. There needs to be a serious discussion around this knowledge/expertise bottle-neck.

**Reply by dadiorchen** [reply]
Is it possible the local name phase (the No.1) needs support or guidance from the species experts? In this case, the experts will interfere with the determination of the local name, for example, the are using the wrong name to attach to a tree? If this is true, we can use the weight rule to make sure the local name is used correctly.

**Reply by camwebb** [reply]
The way I always saw this was that Phase 1 is fast and generally easy and will not require a botanist to help. The organization users mint a set of local names for their project, given what they have in their greenhouse. Any names will do. Then they apply the names to the photos in the admin panel. 

Phase 2: Once there is a sizeable number of images for each taxon, a GS botanist can examine a sample of images and try to make a link to a GS morphotype name. This will be much faster the the current process I use: I have had to search through thousands of unidentified images at a site, creating collections of similar taxa, then trying to match these to global morphotypes. 

The advantages of the new system are i) the generation by users of same-taxon sets of images, and ii) the provision by users of a name; the local name is often (but not always) useful in identifying and speeding up the matching to global morphotypes. In the past we've had to approach the organization to try to get a list of nursery species. There is no standard mechanisms for this direct communication and it's been hit-or-miss.

**Reply by dadiorchen** [reply]
@camwebb did you receive my message in slack? A personal question, what's the best botanical garden in Indonesia :D

**Reply by camwebb** [reply]
I just replied via email.

**Comment by lorepostman** [comment]
Thanks  @camwebb for the tag. I agree this will require funding. The good news is that we (Greenstand) are building a big strategy to be funded by grants, which we've already begun applying for. There's a big overarching strategy (~$4m total), and we'd want to make sure the project to build the digital herbarium system is staffed and funded within that strategy. 
 If the grants come through, the launch date would be January 2024. The bulk of the work would be done in 24 months (wrapping up in Dec. 2025). Thanks in advance for your help outlining the scientific support needed to be put in place for this to be successful!!
